src/main/java/jenkins/plugins/logstash/persistence/ElasticSearchDao.java

```diff
@@ -149,12 +165,38 @@ public String getMimeType() {
   public void setMimeType(String mimeType) {
     this.mimeType = mimeType;
   }
+
+  public KeyStore getCustomKeyStore() {
+    return this.customKeyStore;
+  }
+
+  public void setCustomKeyStore(KeyStore customKeyStore) {
+    this.customKeyStore = customKeyStore;
+
+    // This function sets the state of HttpClientBuilder,
+    // hence can safely be called multiple times if keystore is changed
+    setClientBuilderSSLContext();
+  }
   
   String getAuth()
   {
     return auth;
   }
 
+  private void setClientBuilderSSLContext() {
```

Comment:
this is quite long, how about moving this and createSSLContext to a separate class/helper method that would depend on customKeyStore and clientBuilder?

Reply:
Resolved - I've moved the setting and creation to a separate `SSLHelper` class